### PR TITLE
story 1250,1269

### PR DESCRIPTION
**Pull request by GSuma21**

### CODE REVIEW

File: projects/lib-project/src/lib/lib-project.service.ts [hunk 1]
```diff
@@ -97,14 +97,19 @@ export class LibProjectService {
...
         })
       }else{
-        this.sendForReview({},this.projectData.id).subscribe((res:any) =>{
-          let data = {
-            message : res.message,
-            class : 'success'
-          }
-          this.toastService.openSnackBar(data)
-          this.router.navigate([SUBMITTED_FOR_REVIEW]);
+        this.getResolveComment().subscribe((res) => {
```
Comment (ramkumar-pacewisdom) [inline projects/lib-project/src/lib/lib-project.service.ts:100]: how this is unsubscribed ?

File: projects/lib-project/src/lib/lib-project.service.ts [hunk 2]
```diff
@@ -97,14 +97,19 @@ export class LibProjectService {
...
+          this.utilService.updateComment(this.projectData.id,res).subscribe((res:any)=>{
```
Comment (ramkumar-pacewisdom) [inline projects/lib-project/src/lib/lib-project.service.ts:101]: same here ?

File: projects/lib-project/src/lib/lib-project.service.ts [hunk 3]
```diff
@@ -97,14 +97,19 @@ export class LibProjectService {
...
+            this.sendForReview({},this.projectData.id).subscribe((res:any) =>{
```
Comment (ramkumar-pacewisdom) [inline projects/lib-project/src/lib/lib-project.service.ts:102]: here ?

File: projects/lib-project/src/lib/lib-project.service.ts [hunk 4]
```diff
@@ -271,4 +278,20 @@ export class LibProjectService {
     return this.httpService.get( this.Configuration.urlConFig.CERTIFICATE.LIST)
   }
 
+  getResolveComment(){
```
Comment (ramkumar-ns-dev) [inline projects/lib-project/src/lib/lib-project.service.ts]: why we need this function ?

File: projects/lib-shared-modules/src/lib/components/comments-box/comments-box.component.ts [hunk 5]
```diff
@@ -122,14 +119,16 @@ export class CommentsBoxComponent implements OnInit, OnDestroy {
   }
 
 
-  saveComment() {
-    console.log(this.quillInput)
-    this.chatFlag = !this.chatFlag;
+  saveComment(save :any ="") {
```
Comment (ramkumar-ns-dev) [inline projects/lib-shared-modules/src/lib/components/comments-box/comments-box.component.ts]: why parameter like this ?

**Review by ramkumar-pacewisdom — CHANGES_REQUESTED** [review]

**Review by GSuma21 — COMMENTED** [review]
optimised get comments and change status functions and parameter

File: projects/lib-project/src/lib/lib-project.service.ts [hunk 6]
```diff
@@ -97,14 +97,19 @@ export class LibProjectService {
...
-          let data = {
-            message : res.message,
-            class : 'success'
-          }
-          this.toastService.openSnackBar(data)
-          this.router.navigate([SUBMITTED_FOR_REVIEW]);
+        this.getResolveComment().subscribe((res) => {
+          this.utilService.updateComment(this.projectData.id,res).subscribe((res:any)=>{
+            this.sendForReview({},this.projectData.id).subscribe((res:any) =>{
+              let data = {
```
Comment (ramkumar-ns-dev) [inline projects/lib-project/src/lib/lib-project.service.ts]: just pass the object as parameter

File: projects/lib-project/src/lib/lib-project.service.ts [hunk 7]
```diff
@@ -271,4 +266,38 @@ export class LibProjectService {
...
+          } else {
+            comment.status = 'OPEN';
+          }
+        });
+        return comments;
+      })
+    );
+  }
+  
+  getOpenCommentList(): Observable<any> {
```
Comment (ramkumar-ns-dev) [inline projects/lib-project/src/lib/lib-project.service.ts]: change the name into "getcommentsListAsOpen()"

**Review by ramkumar-pacewisdom — CHANGES_REQUESTED** [review]

**Review by ramkumar-pacewisdom — APPROVED** [review]

**Pull request merged**